File: core-node/code/nodejs-debug/4-final-solution-exercise-built-debugger/fibonacci.js
// Aplicación Node.js para calcular el número N de la sucesión de Fibonacci.
/*  La sucesión de Fibonacci es una serie de números en la que cada número es la suma de los dos
    anteriores. */
// Para ejecutar este script, escribe en la terminal: node fibonacci.js
function fibonacci(n) {
  let n1 = 0;
  let n2 = 1;
  let sum = 0;

  if (n === 0) return n1;
  if (n === 1) return n2;
  

  for (let i = 2; i <= n; i++) {
    sum = n1 + n2;
    n1 = n2;
    n2 = sum;
  }
  return n2;
}

const result = fibonacci(5);
console.log(result); // 5

File: core-node/code/node-dependencies/Ejercicios realizados/Eli-ejercicio-tema-4/DEPURAR_NODEjs/nodejs-debug/fibonacci.c
#include <stdio.h>

void fibonacci(int n, int *result) {
    int n1 = 0;
    int n2 = 1;
    int sum = 0;

    if (n == 0) {
        *result = n1;
        return;
    }

    if (n == 1) {
        *result = n2;
        return;
    }

    for (int i = 2; i <= n; i++) {
        sum = n1 + n2;
        n1 = n2;
        n2 = sum;
    }

    *result = n2;
}

int main() {
    int n = 5;
    int result;
    fibonacci(n, &result);
    printf("Fibonacci of %d is %d\n", n, result);
    return 0;
}